homeassistant/components/zwave_js/climate.py

```diff
@@ -162,6 +162,15 @@ def __init__(
             add_to_watched_value_ids=True,
         )
         self._set_modes_and_presets()
+        self._supported_features = SUPPORT_PRESET_MODE
+        # If any setpoint value exists, we can assume temperature
+        # can be set
+        if any(self._setpoint_values.values()):
+            self._supported_features |= SUPPORT_TARGET_TEMPERATURE
+        if HVAC_MODE_HEAT_COOL in self.hvac_modes:
```

Comment:
Is it possible to know how many setpoints each possible mode provides, ie if there's a high and a low setpoint for that mode or just a single temperature setpoint?

Reply:
yes, the `THERMOSTAT_MODE_SETPOINT_MAP` constant in the library maps `ThermostatMode` to a setpoint list, and `ZW_HVAC_MODE_MAP` in HA maps `ThermostatMode` to an HA mode. So we could work backwards to map HA mode to setpoint list. Is the thought that we rely on that to determine which targets to show?

Reply:
Yes, we want to show high and low target temp if the mode has the two corresponding setpoints but only target temp if the mode only has one setpoint. Looks like it's only `ThermostatMode.AUTO` and `ThermostatMode.AWAY` that have two setpoints.

Reply:
So the current solution is already correct.

Reply:
uhh I don't think it is based on that statement. low will have always have a value with this implementation.

Reply:
Ok, but we don't set supported features to range unless there's a mode with two setpoints. So that part is correct.

Reply:
I mean this PR is correct. We can address the temperature attributes separately if needed.